WebRoot/js/zimbraAdmin/common/ZaTree.js

```diff
@@ -524,10 +524,21 @@ function(item, ev) {
 
         this._selectedItems.add(item);
 
-		if (item._setSelected(true)) {
-            this._updateHistory(item, true, isShowInHistory);
-			this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
-		}
+        if (item._setSelected(true)) {
+            if (item._text === ZaMsg.NAD_Tab_HSM) {
+                if (typeof ZaHSM === 'undefined') {
+                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicense);
+                } else if (ZaHSM && ZaHSM.checkstoragefeaturenabled()) {
+                    this._updateHistory(item, true, isShowInHistory);
+                    this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
+                } else {
+                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicense);
+                }
```

Comment:
We can refactor it as lines 530 and 535 are the same.
```suggestion
                if (ZaHSM && ZaHSM.checkstoragefeaturenabled()) {
                    this._updateHistory(item, true, isShowInHistory);
                    this._notifyListeners(DwtEvent.SELECTION, [item], DwtTree.ITEM_SELECTED, ev, this._selEv);
                } else {
                    ZaApp.getInstance().getCurrentController().popupErrorDialog(ZaMsg.ERROR_Feature_Not_Licensed + "<br>" + ZaMsg.INFO_gotoLicense);
                }
```

Reply:
done